Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -26,6 +26,9 @@ public enum VersionSetSpecifier: Equatable {
     /// A non-empty range of version.
     case range(Range<Version>)
 
+    /// The unit or singleton set.
+    case unit(Version)
```

Comment:
That makes sense for SHA/pinned dependencies but I think the unit version makes sense in general because you can specify exact versions in manifest which we should be resolving (as opposed to a pinned state where whatever is present should be picked ignoring other constraints)